Simperium/src/main/java/com/simperium/android/CredentialsActivity.java

```diff
@@ -41,7 +40,11 @@
 import static com.simperium.android.AuthenticationActivity.EXTRA_IS_LOGIN;
 
 public class CredentialsActivity extends AppCompatActivity {
-    private static final Pattern PATTERN_PASSWORD = Pattern.compile("^(.){4,}$", Pattern.DOTALL);
+    private static final int PASSWORD_LENGTH_LOGIN = 4;
+    private static final int PASSWORD_LENGTH_SIGNUP = 6;
+
+    private static final Pattern PATTERN_PASSWORD_LOGIN = Pattern.compile("^(.){" + PASSWORD_LENGTH_LOGIN + ",}$", Pattern.DOTALL);
+    private static final Pattern PATTERN_PASSWORD_SIGNUP = Pattern.compile("^(.){" + PASSWORD_LENGTH_SIGNUP + ",}$", Pattern.DOTALL);
```

Comment:
Since we accept any character other than whitespace at the moment, those password patterns are effectively the same as a length check.  I chose to use the regular expression rather than the `.length()` method so that we only need to update the `.` part of the regular expression when we want to limit the passwords to certain characters.  I can change that to use `.length()` if it's better.

Reply:
I updated from regular expression to the `.length()` method in 236b87f.